tests/tests.rs

```diff
@@ -322,6 +322,35 @@ fn merge() {
 }
 
 #[test]
+fn merge_by() {
+    let odd : Vec<(u32, &str)> = vec![(1, "hello"), (3, "world"), (5, "!")];
+    let even = vec![(2, "foo"), (4, "bar"), (6, "baz")];
+    let expected = vec![(1, "hello"), (2, "foo"), (3, "world"), (4, "bar"), (5, "!")];
+    let results = odd.iter().merge_by(even.iter(), |a, b|{ a.0.cmp(&b.0)});
+    for (r, e) in results.zip(expected) {
```

Comment:
there's assert_iters_equal for this if you want to. It actually catches something here -- result and expected are different length! :)

     assert_iters_equal(results, expected.iter());